server/rest/src/test/java/org/infinispan/rest/resources/security/SimpleSecurityDomain.java

```diff
@@ -1,24 +1,30 @@
 package org.infinispan.rest.resources.security;
 
+import java.util.HashMap;
+import java.util.Map;
+
 import javax.security.auth.Subject;
 
 import org.infinispan.rest.authentication.SecurityDomain;
 
 /**
- * Security domain that returns always the same subject
+ * Security domain that supports a simple map of subjects
  */
 public class SimpleSecurityDomain implements SecurityDomain {
 
-   private final Subject subject;
+   private final Map<String, Subject> subjects;
 
-   public SimpleSecurityDomain(Subject subject) {
-      this.subject = subject;
+   public SimpleSecurityDomain(Subject... subjects) {
+      this.subjects = new HashMap<>(subjects.length);
+      for (Subject subject : subjects) {
+         this.subjects.put(subject.getPrincipals().iterator().next().getName().toLowerCase(), subject);
+      }
    }
 
    @Override
    public Subject authenticate(String username, String password) throws SecurityException {
       if (username.equals(password)) {
```

Comment:
wait!? is this correct? username equals to password?

Reply:
It's a test :)

Reply:
I'll blame the coffee. it takes time to kick in :rofl: